### PR TITLE
Use file:/// style uris when passing platform to the compiler.

### LINKED ISSUE #116495: [tool] on Windows `flutter test --platform=chrome` crashes with Dart Kernel error
## Steps to Reproduce

Using a Windows computer From flutter/flutter repository root:

1. `cd packages/flutter`
2. `flutter test --platform=chrome  .\test\widgets\router_test.dart`

**Expected results:**

The test should run.

**Actual results:**

The tool crashes with this:

```
Error: Error when reading 'c:/code/flutter/flutter/bin/cache/flutter_web_sdk/kernel/ddc_outline.dill': StandardFileSystem only supports file:* and data:* URIs
```

I checked and `c:/code/flutter/flutter/bin/cache/flutter_web_sdk/kernel/ddc_outline.dill` does exist. Perhaps it's complaining about the `/` in a Windows path?

I reproduced it in 19ef0f9a59e0111947817916657fa62985b33c7f

I'm suspecting this started with https://github.com/flutter/engine/pull/37940, but I don't have a proof.

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
This fixes https://github.com/flutter/flutter/issues/116495

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/test/web_test_compiler_test.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import 'package:file/memory.dart';
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/test/web_test_compiler_test.dart:1]: nit: license header

Comment (eyebrowsoffire) [reply packages/flutter_tools/test/general.shard/test/web_test_compiler_test.dart:5]: Somehow I manage to forget this every single time I make a file.

File: packages/flutter_tools/test/general.shard/test/web_test_compiler_test.dart [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import 'package:flutter_tools/src/build_info.dart';
+import 'package:flutter_tools/src/test/web_test_compiler.dart';
+import 'package:test/expect.dart';
+
+import '../../src/context.dart';
+import '../../src/logging_logger.dart';
+
+void main() {
+  testUsingContext('web test compiler issues valid compile command', () async {
+    final LoggingLogger logger = LoggingLogger();
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/test/web_test_compiler_test.dart:18]: I would use BufferLogger.test() for this

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/general.shard/test/web_test_compiler_test.dart [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/build_info.dart';
+import 'package:flutter_tools/src/test/web_test_compiler.dart';
+import 'package:test/expect.dart';
+
+import '../../src/context.dart';
+
+void main() {
+  testUsingContext('web test compiler issues valid compile command', () async {
```
Comment (yjbanov) [inline packages/flutter_tools/test/general.shard/test/web_test_compiler_test.dart:17]: Consider leaving the link to the issue so a future reader can understand what needs to be valid about the compile command.

**Review by yjbanov — APPROVED** [review]

**Pull request merged**